src/metabase/models/api_key.clj

```diff
@@ -0,0 +1,57 @@
+(ns metabase.models.api-key
+  (:require [crypto.random :as crypto-random]
+            [metabase.util.password :as u.password]
+            [methodical.core :as methodical]
+            [toucan2.core :as t2]))
+
+;; the prefix length, the length of `mb_1234`
+(def ^:private prefix-length 7)
+
+;; the total number of bytes of randomness we generate for API keys
+(def ^:private bytes-key-length 32)
+
+(methodical/defmethod t2/table-name :model/ApiKey [_model] :api_key)
+
+(doto :model/ApiKey
+  (derive :metabase/model)
+  (derive :hook/timestamped?))
+
+(defn prefix
+  "Given an API key, returns the standardized prefix for that API key."
+  [key]
+  (apply str (take prefix-length key)))
+
+(defn- add-prefix [{:keys [unhashed_key] :as api-key}]
+  (cond-> api-key
+    key (assoc :key_prefix (prefix unhashed_key))))
```

Comment:
I'm confused by this `cond->`. `key` is the condition, but it's a function so it'll always be truthy. Can you just use a simple `assoc` here without the `cond->`?

Reply:
Oops, this was a mistake from a rename. The condition should have been `unhashed_key` - we only want to attach a `key_prefix` if someone is trying to set the `unhashed_key`.